instant-xml/src/impls.rs

```diff
@@ -488,6 +488,42 @@ impl<T: ToXml> ToXml for Option<T> {
     }
 }
 
+impl<T: ToXml + ?Sized> ToXml for Box<T> {
+    fn serialize<W: fmt::Write + ?Sized>(
+        &self,
+        field: Option<Id<'_>>,
+        serializer: &mut Serializer<W>,
+    ) -> Result<(), Error> {
+        self.as_ref().serialize(field, serializer)
+    }
+}
+
+impl<'xml, T: FromXml<'xml>> FromXml<'xml> for Box<T> {
+    #[inline]
+    fn matches(id: Id<'_>, field: Option<Id<'_>>) -> bool {
+        T::matches(id, field)
+    }
+
+    fn deserialize<'cx>(
+        into: &mut Self::Accumulator,
+        field: &'static str,
+        deserializer: &mut Deserializer<'cx, 'xml>,
+    ) -> Result<(), Error> {
+        if into.is_some() {
+            return Err(Error::DuplicateValue(field));
+        }
+
+        let mut value = T::Accumulator::default();
+        T::deserialize(&mut value, field, deserializer)?;
+        *into = Some(Box::new(value.try_done(field)?));
```

Comment:
Question, can the try operator be replaced with expect?

Reply:
What do you mean?